optimum/pipelines.py

```diff
@@ -0,0 +1,106 @@
+from typing import Any, Optional, Union
+
+from transformers import (
+    AutoTokenizer,
+    FeatureExtractionPipeline,
+    Pipeline,
+    PreTrainedTokenizer,
+    QuestionAnsweringPipeline,
+    TextClassificationPipeline,
+    TextGenerationPipeline,
+    TokenClassificationPipeline,
+    ZeroShotClassificationPipeline,
+)
+from transformers import pipeline as transformers_pipeline
+from transformers.feature_extraction_utils import PreTrainedFeatureExtractor
+
+from optimum.onnxruntime.modeling_ort import ORTModelForCausalLM
+from optimum.utils import is_onnxruntime_available
+
+
+SUPPORTED_TASKS = {}
+
+if is_onnxruntime_available():
+    from optimum.onnxruntime import (
+        ORTModel,
+        ORTModelForFeatureExtraction,
+        ORTModelForQuestionAnswering,
+        ORTModelForSequenceClassification,
+        ORTModelForTokenClassification,
+    )
+
+    SUPPORTED_TASKS = {
+        "feature-extraction": {
+            "impl": FeatureExtractionPipeline,
+            "class": (ORTModelForFeatureExtraction,) if is_onnxruntime_available() else (),
+            "default": "distilbert-base-cased",
+        },
+        "text-classification": {
+            "impl": TextClassificationPipeline,
+            "class": (ORTModelForSequenceClassification,) if is_onnxruntime_available() else (),
+            "default": "distilbert-base-uncased-finetuned-sst-2-english",
+        },
+        "token-classification": {
+            "impl": TokenClassificationPipeline,
+            "class": (ORTModelForTokenClassification,) if is_onnxruntime_available() else (),
+            "default": "dbmdz/bert-large-cased-finetuned-conll03-english",
+        },
+        "question-answering": {
+            "impl": QuestionAnsweringPipeline,
+            "class": (ORTModelForQuestionAnswering,) if is_onnxruntime_available() else (),
+            "default": "distilbert-base-cased-distilled-squad",
+        },
+        "zero-shot-classification": {
+            "impl": ZeroShotClassificationPipeline,
+            "class": (ORTModelForSequenceClassification,) if is_onnxruntime_available() else (),
+            "default": "facebook/bart-large-mnli",
+        },
+        "text-generation": {
+            "impl": TextGenerationPipeline,
+            "class": (ORTModelForCausalLM,) if is_onnxruntime_available() else (),
+            "default": "distilgpt2",
+        },
+    }
+
+
+def optimum_pipeline(
```

Comment:
What about just `pipeline()`? We are in the optimum namespace, so it seems pretty obvious to me 🤗